designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementApiHttpService.scala

```diff
@@ -0,0 +1,82 @@
+package pl.touk.nussknacker.ui.services
+
+import cats.data.{EitherT, Validated}
+import cats.syntax.all._
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.deployment.ScenarioActionName
+import pl.touk.nussknacker.engine.api.process.{ProcessId, ProcessIdWithName, ProcessName}
+import pl.touk.nussknacker.engine.deployment.CustomActionDefinition
+import pl.touk.nussknacker.restmodel.CustomActionRequest
+import pl.touk.nussknacker.restmodel.validation.PrettyValidationErrors
+import pl.touk.nussknacker.ui.api.ManagementApiEndpoints.ManagementApiError
+import pl.touk.nussknacker.ui.api.ManagementApiEndpoints.ManagementApiError.{NoActionDefinition, NoScenario}
+import pl.touk.nussknacker.ui.api.{BaseHttpService, CustomActionValidationDto, ManagementApiEndpoints}
+import pl.touk.nussknacker.ui.process.ProcessService
+import pl.touk.nussknacker.ui.process.deployment.DeploymentManagerDispatcher
+import pl.touk.nussknacker.ui.security.api.{AuthenticationResources, LoggedUser}
+import pl.touk.nussknacker.ui.validation.CustomActionValidator
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ManagementApiHttpService(
+    authenticator: AuthenticationResources,
+    dispatcher: DeploymentManagerDispatcher,
+    processService: ProcessService
+)(implicit executionContext: ExecutionContext)
+    extends BaseHttpService(authenticator)
+    with LazyLogging {
+
+  private val managementApiEndpoints = new ManagementApiEndpoints(authenticator.authenticationMethod())
+
+  expose {
+    managementApiEndpoints.customActionValidationEndpoint
+      .serverSecurityLogic(authorizeKnownUser[ManagementApiError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (processName, req) =>
+          for {
+            scenarioId <- getScenarioIdByName(processName)
+            scenarioIdWithName = ProcessIdWithName(scenarioId, processName)
+            actionDefinition <- getActionDefinition(scenarioIdWithName, req.actionName)
+            validator = new CustomActionValidator(actionDefinition)
+            resultsDto <- getValidationResultsDto(validator, req)
+          } yield resultsDto
+        }
+      }
+  }
+
+  private def getScenarioIdByName(scenarioName: ProcessName): EitherT[Future, NoScenario, ProcessId] = {
```

Comment:
I extracted a helper trait for that: `ScenarioHttpServiceExtensions` to avoid code duplication. Can you reuse it?